### ISSUE TITLE
Can't reproduce the results with grid features on single GPU

### ISSUE THREAD

**Issue by liuyyy111** [woodfrog/vse_infty#5]
Hi, congratulations on being accepted for your job

I'm confused that I can't reproduce the result with the grid features (BUTD).
Because of the limit of GPU memory, I use a single V100 to run the model, and use the torch.utils.checkpoint to save the memory when finetuning the ResNet.
But the result is not good, has a large margin with the result in the paper, the results on f30k as follow:

INFO:lib.evaluation:calculate similarity time: 0.683558464050293
INFO:lib.evaluation:rsum: 493.9
INFO:lib.evaluation:Average i2t Recall: 88.8
INFO:lib.evaluation:Image to text: 77.1 92.7 96.5 1.0 3.0
INFO:lib.evaluation:Average t2i Recall: 75.9
INFO:lib.evaluation:Text to image: 56.6 82.0 88.9 1.0 8.1

where is the problem?  multiple GPUs? or torch.utils.checkpoint?

**Comment by woodfrog** [comment]
Hi, I didn't try to use ```torch.utils.checkpoint``` to run this model before, but if all other setups were kept the same, this is likely the problem. Did you change any other settings like image resolution, batch size, or the pre-trained BUTD ResNet for the grid-feature experiments? 

BTW, did you train the models with region features to reproduce the results? If models with region features work well, then I believe the problem should be the use of torch.utils.checkpoint.

**Comment by liuyyy111** [comment]
yep, i also train the model with region features, and it works well. 
I just remove 'CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 ' in the trian_grid.sh, without any other changes

**Comment by woodfrog** [comment]
I see. I found [this post](https://discuss.pytorch.org/t/training-with-gradient-checkpoints-torch-utils-checkpoint-appears-to-reduce-performance-of-model/78102/12) in Pytorch Forum discussing similar issues, not sure if it can help.

**Comment by liuyyy111** [comment]
Thank you!!
